functions/src/index.ts

```diff
@@ -1 +1,43 @@
-import * as functions from 'firebase-functions';
\ No newline at end of file
+const serviceAccount = require('../firebase.json');
+const admin = require('firebase-admin');
+const XLSX = require('xlsx');
+const functions = require('firebase-functions');
+
+admin.initializeApp({
+    credential: admin.credential.cert(serviceAccount),
+    databaseURL: "https://testing-gym-data.firebaseio.com", 
+    client_email: "firebase-adminsdk-l2o7o@testing-gym-data.iam.gserviceaccount.com"
+});
+
+const db = admin.firestore();
+const gyms = ['teagle', 'helen_newman', 'noyes', 'appel'];
+
+exports.getData = functions.https.onRequest(async (req: any, res: any) => {
+    const id = req.query.id;
```

Comment:
You should validate this. Also, callables don't use queries - they have data objects.

functions.https.HttpsError can be used for invalid data.